webrtchack/webrtchack.go

```diff
@@ -0,0 +1,83 @@
+// TODO: This needs to also be called for remotes
+package webrtchack
+
+import (
+	"sync"
+
+	"github.com/pion/webrtc/v3"
+	"go.uber.org/multierr"
+	rdkgrpc "go.viam.com/rdk/grpc"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+)
+
+type OnTrackCB func(tr *webrtc.TrackRemote, r *webrtc.RTPReceiver)
+
+// For modules, the grpc connection is over a Unix socket. The WebRTC `PeerConnection` is made
+// separately. The `SharedConn` continues to implement the `rpc.ClientConn` interface by pairing up
+// the two underlying connections a client may want to communicate over.
+type SharedConn struct {
+	*rdkgrpc.ReconfigurableClientConn
+	peerConnMu         sync.RWMutex
+	resourceOnTrackCBs map[resource.Name]OnTrackCB
+	peerConn           *webrtc.PeerConnection
+	logger             logging.Logger
+}
+
+func NewSharedConn(conn *rdkgrpc.ReconfigurableClientConn, peerConn *webrtc.PeerConnection, logger logging.Logger) *SharedConn {
+	sc := &SharedConn{
+		ReconfigurableClientConn: conn,
+		peerConn:                 peerConn,
+		resourceOnTrackCBs:       map[resource.Name]OnTrackCB{},
+		logger:                   logger,
+	}
+	if peerConn != nil {
+		peerConn.OnTrack(func(tr *webrtc.TrackRemote, r *webrtc.RTPReceiver) {
+			name, err := resource.NewFromString(tr.StreamID())
+			if err != nil {
+				logger.Errorf("%p OnTrack called with StreamID: %s which is not able to be parsed into a resource name", sc, tr.StreamID())
+				return
+			}
+			// TODO: Maybe we don't want to hold this lock for the entire duration of the onTrackCB
```

Comment:
I'm assuming the `peerConnMu` is protecting access to this map? I agree with the concern and think we should just release after the map read.